internal/processors/token_transfer.go

```diff
@@ -0,0 +1,199 @@
+package processors
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"time"
+
+	"github.com/stellar/go/asset"
+	"github.com/stellar/go/ingest"
+	ttp "github.com/stellar/go/processors/token_transfer"
+	"github.com/stellar/go/xdr"
+	"github.com/stellar/wallet-backend/internal/indexer/types"
+	cache "github.com/stellar/wallet-backend/internal/store"
+)
+
+type TokenTransferProcessor struct {
+	contractStore     cache.ContractStore
+	eventsProcessor   *ttp.EventsProcessor
+	networkPassphrase string
+}
+
+func NewTokenTransferProcessor(contractStore cache.ContractStore, networkPassphrase string) *TokenTransferProcessor {
+	return &TokenTransferProcessor{
+		contractStore:     contractStore,
+		eventsProcessor:   ttp.NewEventsProcessor(networkPassphrase),
+		networkPassphrase: networkPassphrase,
+	}
+}
+
+func (p *TokenTransferProcessor) ProcessTransaction(ctx context.Context, tx ingest.LedgerTransaction) ([]types.StateChange, error) {
+	ledgerCloseTime := tx.Ledger.LedgerCloseTime()
+	ledgerNumber := tx.Ledger.LedgerSequence()
+	txHash := tx.Hash.HexString()
+
+	// Get events from this specific transaction
+	txEvents, err := p.eventsProcessor.EventsFromTransaction(tx)
+	if err != nil {
+		return nil, fmt.Errorf("processing token transfer events from transaction: %w", err)
+	}
+
+	// Combine fee events and operation events
+	allEvents := append(txEvents.FeeEvents, txEvents.OperationEvents...)
+
+	stateChanges := make([]types.StateChange, 0, len(allEvents))
+	for _, e := range allEvents {
+		meta := e.GetMeta()
+		contractAddress := meta.GetContractAddress()
+
+		// Get operation details if available
+		var operationType *xdr.OperationType
+		var opSourceAccount string
+		if meta.GetOperationIndex() > 0 {
+			// GetOperation expects 0-based index, but meta has 1-based index
+			if op, found := tx.GetOperation(meta.GetOperationIndex() - 1); found {
+				operationType = &op.Body.Type
+				opSourceAccount = op.SourceAccount.Address()
+			}
+		}
+
+		baseChange := p.createBaseStateChange(ledgerNumber, ledgerCloseTime, txHash)
```

Comment:
I'm missing you associating the operationId with the stateChange object, similarly to how you associated the txHash with the stateChange

Reply:
@marcelosalloum Yes I am sill extracting details from the events. Its still WIP